src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/INLWriter.cpp

```diff
@@ -0,0 +1,217 @@
+const std::atomic_bool& INLWriter::getCancel()
+{
+  return m_ShouldCancel;
+}
+
+// -----------------------------------------------------------------------------
+Result<> INLWriter::operator()()
+{
+  // Make sure any directory path is also available as the user may have just typed
+  // in a path without actually creating the full path
+  Result<> createDirectoriesResult = complex::CreateOutputDirectories(m_InputValues->OutputFile.parent_path());
+  if(createDirectoriesResult.invalid())
+  {
+    return createDirectoriesResult;
+  }
+
+  // Make sure any directory path is also available as the user may have just typed
+  // in a path without actually creating the full path
+  std::ofstream fout(m_InputValues->OutputFile, std::ios_base::out | std::ios_base::binary);
+  if(!fout.is_open())
+  {
+    return MakeErrorResult(-74100, fmt::format("Error creating and opening output file at path: {}", m_InputValues->OutputFile.string()));
+  }
+
+  auto imageGeom = m_DataStructure.getDataRefAs<ImageGeom>(m_InputValues->ImageGeomPath);
+
+  auto& featureIds = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->FeatureIdsArrayPath);
+  auto& eulerAngles = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->CellEulerAnglesArrayPath);
+  auto& cellPhases = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->CellPhasesArrayPath);
+  auto& crystalStructures = m_DataStructure.getDataRefAs<UInt32Array>(m_InputValues->CrystalStructuresArrayPath);
+  auto& numFeatures = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->NumFeaturesArrayPath);
+  auto& materialNames = m_DataStructure.getDataRefAs<StringArray>(m_InputValues->MaterialNameArrayPath);
+
+  usize totalPoints = featureIds.getNumberOfTuples();
+
+  SizeVec3 dims = imageGeom.getDimensions();
+  FloatVec3 res = imageGeom.getSpacing();
+  FloatVec3 origin = imageGeom.getOrigin();
+
+  // Write the header, Each line starts with a "#" symbol
+  fout << "# File written from " << COMPLEX_PACKAGE_COMPLETE << "\n";
+  fout << "# X_STEP: " << std::fixed << res[0] << "\n";
+  fout << "# Y_STEP: " << std::fixed << res[1] << "\n";
+  fout << "# Z_STEP: " << std::fixed << res[2] << "\n";
+  fout << "#\n";
+  fout << "# X_MIN: " << std::fixed << origin[0] << "\n";
+  fout << "# Y_MIN: " << std::fixed << origin[1] << "\n";
+  fout << "# Z_MIN: " << std::fixed << origin[2] << "\n";
+  fout << "#\n";
+  fout << "# X_MAX: " << std::fixed << origin[0] + (static_cast<float64>(dims[0]) * res[0]) << "\n";
+  fout << "# Y_MAX: " << std::fixed << origin[1] + (static_cast<float64>(dims[1]) * res[1]) << "\n";
+  fout << "# Z_MAX: " << std::fixed << origin[2] + (static_cast<float64>(dims[2]) * res[2]) << "\n";
+  fout << "#\n";
+  fout << "# X_DIM: " << static_cast<long long unsigned int>(dims[0]) << "\n";
+  fout << "# Y_DIM: " << static_cast<long long unsigned int>(dims[1]) << "\n";
+  fout << "# Z_DIM: " << static_cast<long long unsigned int>(dims[2]) << "\n";
```

Comment:
I don't think the casts here are necessary since `dims` should be `usize` already.